compiler/src/dotty/tools/dotc/core/NullOpsDecorator.scala

```diff
@@ -0,0 +1,147 @@
+package dotty.tools.dotc.core
+
+import dotty.tools.dotc.core.Contexts.Context
+import dotty.tools.dotc.core.Symbols.defn
+import dotty.tools.dotc.core.Types.{AndType, ClassInfo, ConstantType, OrType, Type, TypeBounds, TypeMap, TypeProxy}
+
+/** Defines operations on nullable types. */
+object NullOpsDecorator {
+
+  implicit class NullOps(val self: Type) {
+    /** Is this type a reference to `Null`, possibly after aliasing? */
+    def isNullType(implicit ctx: Context): Boolean = self.isRef(defn.NullClass)
+
+    /** Is self (after widening and dealiasing) a type of the form `T | Null`? */
+    def isNullableUnion(implicit ctx: Context): Boolean = {
```

Comment:
Can we make this faster? It's wasteful to do all the expensive transformation in `normNullableUnion` only to throw it away afterwards.

Reply:
Done

Reply:
We made this faster by making normNullableUnion faster.